docs/okeps/okep-5259-no-overlay.md

```diff
@@ -0,0 +1,598 @@
+  Advertisement feature, the traffic from pods will no longer be SNATed on
+  egress. In the route advertisement feature, SNAT is disabled for all pod
+  egress traffic that leaves the node, regardless of whether the destination
+  route is BGP learnt or not. For users who only want to have no-overlay mode
+  but do not want to expose podIPs to external, this behavior is not ideal.
+  Instead of disabling SNAT for all the pod-to-external traffic, OVN-K shall
+  adjust its SNAT behavior so that pod IPs are exposed for traffic routed via
+  BGP, but SNAT is applied for non-BGP-routed egress traffic.
+
+* For east-west traffic, intra-node traffic (pod-to-pod, pod-to-clusterIP and
+  pod-to-host) remains unchanged, while cross-node traffic will follow the same
+  path as north-south traffic.
+
+### API Details
+
+We introduce a new `layer3.encapsulation` field to be added to the Spec of the
+ClusterUserDefinedNetwork (CUDN) CRD. This
+new field will enable control over the network encapsualtion behavior with the
+following options:
+
+1. **Encapsulation Configuration**:
+   * `encapsulation`: specifies the encapsualtion method (default: `Geneve`)
+   * Supported values: `Geneve` (default) or `None` (no overlay)
+
+The `spec` field of a ClusterUserDefinedNetwork CR is immutable. Therefore, the
+encapsulation configuration cannot be changed once a ClusterUserDefinedNetwork
+CR is created.
+
+#### Example of a layer-3 CUDN with no-overlay mode enabled
+
+A layer-3 CUDN that enables no-overlay mode should look like this:
+
+```yaml
+apiVersion: k8s.ovn.org/v1
+kind: ClusterUserDefinedNetwork
+metadata:
+  name: my-cudn
+spec:
+  namespaceSelector:
+    matchExpressions:
+    - key: kubernetes.io/metadata.name
+      operator: In
+      values: ["red", "blue"]
+  network:
+    topology: Layer3
+    layer3:
+      role: Primary
+      mtu: 1500
+      subnets:
+      - cidr: 10.10.0.0/16
+        hostSubnet: 24
+      encapsulation: None
+```
+
+#### Example of the default network with no-overlay mode enabled
+
+For the cluster default network, the cluster admin shall create a
+ClusterUserDefinedNetwork CR with the reserved name `default` and set
+encapsulation to `None` to enable no-overlay mode before deploying
+OVN-Kubernetes. For this CR, OVN-K shall not create the corresponding
+NetworkAttachementDefinition CR. The subnet field must be aligned with the value
+in `cluster-subnets` parameter. The MTU shall not be larger than the node
+interface MTU.
+
+```yaml
+apiVersion: k8s.ovn.org/v1
+kind: ClusterUserDefinedNetwork
+Metadata:
+  # 'default' is the reserved name to represent the cluster default network
+  name: default
+spec:
+  # use an empty selector to select all namespaces
+  namespaceSelector: {}
+  network:
+    topology: Layer3
+    layer3:
+      role: Primary
+      mtu: 1500
+      subnets:
+      - cidr: 10.244.0.0/16
+        hostSubnet: 24
+      encapsulation: None
+```
+
+### Implementation Details
```

Comment:
I am missing implementation details describing what OVNK components react to the new API and how.

Reply:
I will add.